### PR TITLE
Add custom hasher option

### PR DESCRIPTION
**Pull request by polachok**

### CODE REVIEW

**Comment by AlisdairO** [comment]
Many thanks for the PR - I'll try to get to it over the weekend!

**Comment by AlisdairO** [comment]
Er, I clearly didn't do that.  Apologies - I've recently emigrated and am very behind at the moment.  I'll try to catch up with my PRs asap.

File: src/lib.rs [hunk 1]
```diff
@@ -341,6 +351,20 @@ impl<K: Eq + Hash + Sync + Clone, V: Sync + Clone> CHMSegment<K, V> {
         guard.unlinked(table);
     }
 
+    fn entries(&self) -> Vec<(K, V)> {
+        let mut xs = vec![];
+        let guard = epoch::pin();
+        let table = self.table.load(Relaxed, &guard).unwrap();
```
Comment (AlisdairO) [inline src/lib.rs]: Not sure if you're still interested in this PR given that I've abandoned you for so long - but I'm a bit concerned about these loads.  I think they will need to be Acquire to be safe, unless you have an explanation as to why not?

File: src/lib.rs [hunk 2]
```diff
@@ -341,6 +351,20 @@ impl<K: Eq + Hash + Sync + Clone, V: Sync + Clone> CHMSegment<K, V> {
         guard.unlinked(table);
     }
 
+    fn entries(&self) -> Vec<(K, V)> {
+        let mut xs = vec![];
```
Comment (AlisdairO) [inline src/lib.rs]: Might be worth using with_capacity to reserve some space for this vec.

File: src/lib.rs (hunk 1 above)
Comment (polachok) [reply src/lib.rs]: I think you're right.

File: src/lib.rs (hunk 2 above)
Comment (polachok) [reply src/lib.rs]: Fixed

**Comment by AlisdairO** [comment]
OK, LGTM - would you mind adding a couple of unit tests?  I'd like something to show, say, that the output of entries == the input data into the hashmap.

**Comment by polachok** [comment]
Added a simple test

**Comment by AlisdairO** [comment]
Fantastic, thank you!

**Pull request merged**